From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Wed, 8 Jul 2015 12:20:48 +0100
Subject: ./configure is not actually a configure script, skip

---
 debian/rules | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/debian/rules b/debian/rules
index a6431f2..1ea9f7b 100755
--- a/debian/rules
+++ b/debian/rules
@@ -5,6 +5,8 @@
 
 mflags=prefix=/usr mandir=/usr/share/man
 
+override_dh_auto_configure:
+
 override_dh_auto_build:
 	dh_auto_build -- $(mflags) all
 
