From: "Barak A. Pearlmutter" <barak+git@cs.nuim.ie>
Date: Wed, 27 Jul 2011 20:46:53 +0100
Subject: recombobulate Makefile

---
 Makefile | 29 +++++++++++++++++++++--------
 1 file changed, 21 insertions(+), 8 deletions(-)

diff --git a/Makefile b/Makefile
index f684093..24a08ac 100644
--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@
 # Placed in the Public Domain.
 
 # Change at least this line:
-PREFIX= /u
+PREFIX=$(prefix)
 
 # Base version and Release
 BASE=		20140708
@@ -55,14 +55,21 @@ DEFS=	$(OSDEF) \
 	-DCURSES_RESET
 
 # Where to install the stuff
-BINDIR=	$(PREFIX)/bin
-LIBDIR=	$(PREFIX)/share/s9fes
-MANDIR=	$(PREFIX)/man/man1
+prefix=/usr/local
+exec_prefix=$(prefix)
+bindir=$(exec_prefix)/bin
+datadir=$(prefix)/share
+libdir=$(exec_prefix)/lib
+mandir=$(prefix)/man
+
+BINDIR=$(DESTDIR)$(bindir)
+LIBDIR=$(DESTDIR)$(libdir)
+MANDIR=$(DESTDIR)$(mandir)/man1
 
 # Set up environment to be used during the build process
 BUILD_ENV=	env S9FES_LIBRARY_PATH=.:lib:ext:contrib
 
-default:	s9 s9.image s9.1.gz s9.1.txt lib/syntax-rules.scm \
+default:	s9 s9.image s9.1 lib/syntax-rules.scm \
 		lib/matcher.scm contrib/bottles.scm
 
 all:	default
@@ -77,8 +84,14 @@ s9.o:	s9.c s9-real.c s9.h
 s9.image:	s9 s9.scm s9-real.scm ext/unix.scm ext/curses.scm config.scm
 	$(BUILD_ENV) ./s9 -i - -n $(EXTRA_SCM) -l config.scm -d s9.image
 
-s9.1.gz:	s9.1
-	sed -e "s,@LIBDIR@,$(LIBDIR)," <s9.1 |gzip -9 >s9.1.gz
+%.1: %.1.in
+	sed \
+	 -e "s,@LIBDIR@,$(LIBDIR),g" \
+	 -e "s,@DATADIR@,$(DATADIR),g" \
+	 < $@.in > $@
+
+%.gz: %
+	gzip -9 < $* > $@
 
 unix.o:	ext/unix.c s9.h
 	$(CC) $(CFLAGS) $(DEFS) -I . -o unix.o -c ext/unix.c
@@ -121,7 +134,7 @@ install-s9:	s9 s9.scm s9.image s9.1.gz
 	install -d -m 0755 $(LIBDIR)/help
 	install -d -m 0755 $(MANDIR)
 	install $C -m 0755 s9 $(BINDIR)
-	strip $(BINDIR)/s9
+	echo skip strip $(BINDIR)/s9
 	install $C -m 0644 s9.scm $(LIBDIR)
 	install $C -m 0644 s9.image $(LIBDIR)
 	install $C -m 0644 lib/* $(LIBDIR)
