From: "Barak A. Pearlmutter" <barak+git@cs.nuim.ie>
Date: Wed, 27 Jul 2011 20:45:24 +0100
Subject: maxpathlen for hurd

---
 s9.h |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/s9.h b/s9.h
index fe51dcc..b3591d4 100644
--- a/s9.h
+++ b/s9.h
@@ -74,6 +74,11 @@
  #endif
 #endif
 
+#ifndef MAXPATHLEN
+ /* MAXPATHLEN not defined on Hurd, as it is "bogus." */
+ #define MAXPATHLEN 4096
+#endif
+
 /*
  * Tell later MSC compilers to let us use the standard CLIB API.
  * Blake McBride < b l a k e  at  m c b r i d e . n a m e >
-- 
