From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Mon, 19 Dec 2016 15:19:10 +0000
Subject: x32 pointer size

---
 s9core.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/s9core.h b/s9core.h
index 6aa7686..69e2a84 100644
--- a/s9core.h
+++ b/s9core.h
@@ -142,6 +142,12 @@
    #ifdef __x86_64
     #define S9_BITS_PER_WORD_64
    #endif
+   #ifdef __ILP32__
+    #define S9_BITS_PER_WORD_32
+   #endif
+   #ifdef __LP64__
+    #define S9_BITS_PER_WORD_64
+   #endif
   #endif
  #endif
 #endif
