From: "Barak A. Pearlmutter" <barak+git@cs.nuim.ie>
Date: Fri, 17 Jan 2014 10:18:26 +0000
Subject: fread checks

---
 s9.c | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/s9.c b/s9.c
index d6dba36..ecdd556 100644
--- a/s9.c
+++ b/s9.c
@@ -4842,7 +4842,9 @@ int load_image(char *p) {
 	f = fopen(p, "rb");
 	if (f == NULL)
 		return -1;
-	fread(&m, sizeof(m), 1, f);
+	if (fread(&m, sizeof(m), 1, f) != 1) {
+		return -1;
+	}
 	if (memcmp(m.id, "S9", 2)) {
 		error("error in image file (magic match failed)", name);
 		ok = 0;
@@ -4870,8 +4872,12 @@ int load_image(char *p) {
 		ok = 0;
 	}
 	memset(Tag, 0, Cons_pool_size);
-	fread(&image_nodes, sizeof(int), 1, f);
-	fread(&image_vcells, sizeof(int), 1, f);
+	if (fread(&image_nodes, sizeof(int), 1, f) != 1) {
+		return -1;
+	}
+	if (fread(&image_vcells, sizeof(int), 1, f) != 1) {
+		return -1;
+	}
 	while (image_nodes > Cons_pool_size) {
 		if (	Memory_limit_kn &&
 			Cons_pool_size + Cons_segment_size > Memory_limit_kn
@@ -4895,7 +4901,9 @@ int load_image(char *p) {
 	v = Image_vars;
 	i = 0;
 	while (v[i]) {
-		fread(v[i], sizeof(cell), 1, f);
+		if (fread(v[i], sizeof(cell), 1, f) != 1) {
+			return -1;
+		}
 		i++;
 	}
 	if (	ok &&
